api.go

```diff
@@ -13,6 +13,8 @@ import (
 type API struct {
 	settingEngine *SettingEngine
 	mediaEngine   *MediaEngine
+	interceptors  []Interceptor
```

Comment:
What do you think of the following?

* Do `settingEngine.interceptors`
* On PeerConnection creation pluck instantiate the chain and do it as `PeerConnection.interceptor`
* If `len(a.interceptors) == 0` use `NoOpInterceptor`. I would like to drop all the `if a.interceptor == nil` checks, eventually we are going to miss one.

Reply:
I removed it from settingEngine for two reasons:
a. The Interceptor factory seemed unnecessarily confusing.
b. See: https://github.com/pion/webrtc-v3-design/blob/interceptor/interceptor/transportcc.go#L26 Some interceptors will need to register stuff with the MediaEngine. That means, as MediaEngine is per PeerConnection, you will need to call Configure** methods for every PeerConnection anyway. So putting the RegisterDefaultInterceptors into SettingEngine would be confusing I think. 

> If len(a.interceptors) == 0 use NoOpInterceptor. I would like to drop all the if a.interceptor == nil checks, eventually we are going to miss one.

Good idea, will do that.